### DISCUSSION TITLE
GroBoost integration

### DISCUSSION THREAD

**Discussion by NeoIsBackCZ** [johanmeijer/grott#355] [Ideas]
First of all thank you for your excelent work! 

I have ShineLanBox, hybrid interter SPH4~10k TL3 BH and also GroBoost (I use it to heat water when I have more power then I need), both are connected to ShineLanBox via RF. I get all values from my inverter in Grott but no values for GroBoost. Is there any way you can add values from GroBoost and maybe also way to control GroBoost via Grott?

I am willing to do some testing if needed.

**Comment by johanmeijer** [comment]
Ah! interesting, a new device. I have not seen this one before. But seems to be a valuable add-on to the installation.

If it is connected to the shinelink and sends data to the growth, we should be able to see and publish that with grott. 

We first have to find the record in the grott log (look for an unknown layout), and then we have to find out how the record layout looks like. 
It would be nice if we could find documentation about it. Otherwise, we have to do some reversed engineering (need info from growatt together with data record from the same time andsee if we can match that.

Are you also able to control the device from the growatt website (or mobile app)?

**Reply by NeoIsBackCZ** [reply]
Yes, device can  be controlled from website or app. 

I  run grott as a docker image on Synology NAS.  Attached is zipped html logfile from docker. 
[grottlog.zip](https://github.com/johanmeijer/grott/files/11826702/grottlog.zip)

EDIT: removed serial numbers from log file.

**Reply by felixhix** [reply]
If there is interest in a contribution or support from my side: I am also owner of a Groboost since today (with running Grott installation under Alpine). The rest of the setup is SPH10k TL3 UP and THOR EV Charger 22kw. If debug or raw messages are needed, I can provide them.

**Comment by johanmeijer** [comment]
Ok. I am going to look into it and see if we can find the groboost records.

It might take some time while I am abroad for work/holidays the comming weeks.

**Reply by NeoIsBackCZ** [reply]
Ok, let me know if you need anything else from me.

**Reply by NeoIsBackCZ** [reply]
Just want to ask you if you had any time to look into this?

Thanks

**Comment by johanmeijer** [comment]
Thanks for remembering me. I have not look at it yet. 

I see a new type of record in your log. That might be from the Groboot. The challenge will be to reverse engineer that if we do not have same kind of protocol / register description. But we can try. 

Is the serial nummer of the Groboot: 6DAK0BLC038 and its device address in the shinelink 80? (I think 81 is your inverter). 

If we want to create an API we have to give the commands you want to give via the Growatt website and then use the grott log to find the format of the command. Are you able to do that?

**Reply by NeoIsBackCZ** [reply]
Serial number is without 6 at the start, so DAK0BLC038, device address is 80, that's correct. I can send commands via the Growatt website, so finding format of the command in grott log should be possible, too.

**Comment by NeoIsBackCZ** [comment]
Just want to check if you were able to make any progress with possible GroBoost integration?

**Reply by Suzukithunder** [reply]
I am listening, have the same product. Willing to work on it :)

**Comment by johanmeijer** [comment]
No progress yet. Some questions. Is your datalogger (shinelink) device: NAC9BL30JF and is  6DAK0BLC03 your inverter or the GroBoost? 

The challenge is I do no have the register layout documentation of the GroBoost so it will be a real puzzle to decipher 

What do we expect to see and/or change from the GroBoost?

**Reply by NeoIsBackCZ** [reply]
Yes, datalogger is NAC9BL30JF and Groboost is DAK0BLC038. I have connected one water heater to GroBoost (you can have more) and I would like to see it's current load/power, current temperature, status (on/off), set temperature.

**Reply by Suzukithunder** [reply]
Ill be happy to assist checking the file as well. Can also provide a file myself with all 3 phases used.

**Comment by NeoIsBackCZ** [comment]
Just another kind reminder if there has been any progress with implementing GroBoost support. As previously stated I'll be more than happy to assist with deciphering register layout.

**Comment by johanmeijer** [comment]
Yes it is on the list. At this moment I work at Grott 3.0 (several enhancements on Grott and Grottserver and the integration of both). That take more spare time then expected. 

Growatt boost is next on the list. This will be a challenge while I have not found a protocol description for this device. So it will be total reversed engeneering. While I do na have a Growatt Boost I have to totally relay on your input.
